File: Valid Anagram/Valid_Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }
        int[] alphFreq = new int [26];
        for(char c : s.toCharArray()){
            alphFreq[c - 'a']++;
        }
        for (char c : t.toCharArray()){
            alphFreq[c - 'a']--;
            if (alphFreq[c - 'a'] < 0){
                return false;
            }
        }
        return true;
    }
}

File: Coding Problems & My Solutions/Valid Anagram/Valid_Anagram.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        var dict: [Character:Int] = [:]
        for char in Array(s) {
            var charCount = dict[char] ?? 0
            dict[char] = charCount + 1
        }
        for char in Array(t) {
            var charCount = dict[char] ?? 0
            dict[char] = charCount - 1
        }
        for (key, value) in dict {
            if value != 0 {
                return false
            }
        }
        return true
    }
}